app/src/test/java/de/westnordost/streetcomplete/quests/smoking/AddSmokingTest.kt

```diff
@@ -0,0 +1,220 @@
+    }
+
+    @Test fun `not applicable to recently solved smoking quest`() {
+        assertFalse(questType.isApplicableTo(node(tags = mapOf(
+            "amenity" to "cafe",
+            "smoking" to "isolated",
+        ))))
+    }
+
+    @Test fun `applicable to old places tagged with smoking-date`() {
+        assertTrue(questType.isApplicableTo(node(tags = mapOf(
+            "amenity" to "food_court",
+            "smoking" to "yes",
+            "smoking:date" to "2014-10-10"
+        ))))
+    }
+
+    /* we assume that seating is not present if not indicated for bakery and similar */
+    @Test fun `not applicable to bakery without indicated seating`() {
+        assertFalse(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+        ))))
+    }
+
+    @Test fun `not applicable to bakery without any seating`() {
+        assertFalse(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+            "indoor_seating" to "no",
+            "outdoor_seating" to "no",
+        ))))
+    }
+
+    @Test fun `applicable to bakery with indoor seating`() {
+        assertTrue(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+            "indoor_seating" to "bar_table",
+        ))))
+    }
+
+    @Test fun `applicable to bakery with only indoor seating`() {
+        assertTrue(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+            "indoor_seating" to "yes",
+            "outdoor_seating" to "no",
+        ))))
+    }
+
+    @Test fun `not applicable to bakery without indoor seating`() {
+        assertFalse(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+            "indoor_seating" to "no",
+        ))))
+    }
+
+    @Test fun `applicable to bakery with outdoor seating`() {
+        assertTrue(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+            "outdoor_seating" to "terrace",
+        ))))
+    }
+
+    @Test fun `applicable to bakery with only outdoor seating`() {
+        assertTrue(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+            "indoor_seating" to "no",
+            "outdoor_seating" to "yes",
+        ))))
+    }
+
+    @Test fun `not applicable to bakery without outdoor seating`() {
+        assertFalse(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+            "outdoor_seating" to "no",
+        ))))
+    }
+
+
+    @Test fun `applicable to bakery with both seatings`() {
+        assertTrue(questType.isApplicableTo(node(tags = mapOf(
+            "shop" to "bakery",
+            "indoor_seating" to "bar_table",
+            "outdoor_seating" to "yes",
+        ))))
+    }
+
+    /* nighclubs etc. may have outdoor smoking areas even if no seating is present */
+    @Test fun `applicable to nightclub without any seating`() {
+        assertFalse(questType.isApplicableTo(node(tags = mapOf(
```

Comment:
The comment above suggests that this should be `assertTrue`.

```suggestion
        assertTrue(questType.isApplicableTo(node(tags = mapOf(
```